types/npmcli__map-workspaces/index.d.ts

```diff
@@ -0,0 +1,28 @@
+// Type definitions for @npmcli/map-workspaces 3.0
+// Project: https://github.com/npm/map-workspaces#readme
+// Definitions by: Emily M Klassen <https://github.com/forivall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// Minimum TypeScript Version: 4.3
+
+import { PackageJson } from '@npmcli/package-json';
+import { PackageLock } from '@npmcli/arborist';
+import { IOptions as GlobOptions } from 'glob';
+
+export = mapWorkspaces;
+
+declare function mapWorkspaces(opts: mapWorkspaces.Options): Promise<Map<string, string>>;
```

Comment:
```suggestion
/** @async */
declare function mapWorkspaces(opts: mapWorkspaces.Options): Promise<Map<string, string>>;
```
https://github.com/npm/map-workspaces/blob/main/lib/index.js#L73

Reply:
Why add this comment? I'm not sure how it would impact the types...

Reply:
there is no impact, you're right. TS declarations lacks `async` keyword, native to TS, so adding this IMO helps to properly denote method written with async/await support in mind (not just returning a promise like object)

Reply:
I think that issue matters more at the actual code execution level, w.r.t async stack trace support, node versions that can use the module, and such. Otherwise, async/await/promises are handled transparently, and although the source function is written with async/await, I dont _think_ it has any impact on the typings - ie. the `async` keyword isnt even shown in the typing hint in vscode (and probably not the playground either)
<img width="537" alt="Screen Shot 2022-10-24 at 1 47 27 PM" src="https://user-images.githubusercontent.com/760204/197626430-0b1051d7-df18-4626-a5f0-b59369fdfe5a.png">